src/EventGrid/EventGrid/SystemTopic/NewAzureEventGridSystemTopic.cs

```diff
@@ -0,0 +1,209 @@
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using EventGridModels = Microsoft.Azure.Management.EventGrid.Models;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    [Cmdlet(
+        "New",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridSystemTopic",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = TopicNameParameterSet),
+    OutputType(typeof(PSSystemTopic))]
+
+    public class NewAzureEventGridSystemTopic : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.ResourceGroupNameHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasResourceGroup)]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = EventGridConstants.TopicNameHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ResourceNameCompleter("Microsoft.EventGrid/systemTopics", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        [Alias("Name")]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = false,
+            Position = 3,
+            HelpMessage = EventGridConstants.TopicNameHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ResourceNameCompleter("Microsoft.EventGrid/systemTopics", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceId")]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = false,
+            Position = 4,
+            HelpMessage = EventGridConstants.TopicNameHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ResourceNameCompleter("Microsoft.EventGrid/systemTopics", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceName")]
+        public string ResourceName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = false,
+            Position = 5,
+            HelpMessage = EventGridConstants.TopicNameHelp,
+            ParameterSetName = SystemTopicNameParameterSet)]
+        [ResourceNameCompleter("Microsoft.EventGrid/systemTopics", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceType")]
+        public string ResourceType { get; set; }
```

Comment:
The name of ResourceType is ResourceType. It needn't an alias. Please check all parameters.